scalar.go

```diff
@@ -19,11 +19,11 @@ func NewID(id ...string) *ID {
 
 func (s ID) GetGraphQLType() string { return "ID" }
 
-func (s *ID) MarshalJSON() ([]byte, error) {
-	if *s == "" {
+func (s ID) MarshalJSON() ([]byte, error) {
```

Comment:
This was changed because it is considered a mistake to mix pointers and non-pointers in receiver functions. Our other `MarshalJSON` functions are not using pointers. 

> First, and most important, does the method need to modify the receiver? If it does, the receiver must be a pointer.
> Second is the consideration of efficiency. If the receiver is large, a big struct for instance, it will be much cheaper to use a pointer receiver.
> For types such as basic types, slices, and small structs, a value receiver is very cheap so unless the semantics of the method requires a pointer, a value receiver is efficient and clear.

https://go.dev/doc/faq#methods_on_values_or_pointers

---

**This change requires the test case for scalar_test.go to change.**